src/components/Publish/_validation.ts

```diff
@@ -1,31 +1,73 @@
 import { MAX_DECIMALS } from '@utils/constants'
 import * as Yup from 'yup'
 import { getMaxDecimalsValidation } from '@utils/numbers'
+import { validateFieldSchaclSchema } from '@utils/schaclSchema'
 import { FileInfo } from '@oceanprotocol/lib'
 
 // TODO: conditional validation
 // e.g. when algo is selected, Docker image is required
 // hint, hint: https://github.com/jquense/yup#mixedwhenkeys-string--arraystring-builder-object--value-schema-schema-schema
-
 const validationMetadata = {
   type: Yup.string()
     .matches(/dataset|algorithm/g, { excludeEmptyString: true })
     .required('Required'),
   name: Yup.string()
-    .min(4, (param) => `Title must be at least ${param.min} characters`)
-    .required('Required'),
+    .required('Required')
+    .test(async (value, { path, createError }): Promise<any> => {
+      if (!value) return
```

Comment:
think we can move this logic in a separate method and just call it in the` test()` . we have the same logic repeating in quite a few places here.